### PR TITLE
[DOC]: Add explanation about DataFrame methods use all Categories

### LINKED ISSUE #36740: DOC: Summing by level on a category index produce extra columns.
Generate a dataframe:
```
import pandas as pd
import numpy as np
column_cat = pd.CategoricalDtype(['One', 'Two', 'Three', 'Four'], ordered=True)
groups = list('AABBB')
numbers = pd.Categorical(['One', 'Two','One', 'Two', 'Three'], 
                         categories=column_cat.categories, ordered=True)
d = pd.DataFrame(data=np.arange(15).reshape(3, 5), 
                 columns=pd.MultiIndex.from_arrays([groups, numbers]))
```

<table border="1" class="dataframe">
  <thead>
    <tr>
      <th></th>
      <th colspan="2" halign="left">A</th>
      <th colspan="3" halign="left">B</th>
    </tr>
    <tr>
      <th></th>
      <th>One</th>
      <th>Two</th>
      <th>One</th>
      <th>Two</th>
      <th>Three</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>1</td>
      <td>2</td>
      <td>3</td>
      <td>4</td>
    </tr>
    <tr>
      <th>1</th>
      <td>5</td>
      <td>6</td>
      <td>7</td>
      <td>8</td>
      <td>9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>10</td>
      <td>11</td>
      <td>12</td>
      <td>13</td>
      <td>14</td>
    </tr>
  </tbody>
</table>

Now  do `d.sum(axis=1, level=1)`, get the following result:

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>One</th>
      <th>Two</th>
      <th>Three</th>
      <th>Four</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>2</td>
      <td>4</td>
      <td>4</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>12</td>
      <td>14</td>
      <td>9</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>22</td>
      <td>24</td>
      <td>14</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

Which produces a non existing column `Four`.

pandas 1.1.1, pandas 1.1.2

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #36740

### CODE REVIEW

File: doc/source/user_guide/categorical.rst [hunk 1]
```diff
@@ -618,6 +618,14 @@ even if some categories are not present in the data:
     s = pd.Series(pd.Categorical(["a", "b", "c", "c"], categories=["c", "a", "b", "d"]))
     s.value_counts()
 
+``DataFrame`` methods like :meth:`DataFrame.sum` also show "unused" categories.
+
+.. ipython:: python
+
+    columns = pd.Categorical(['One','One', 'Two'], categories=['One', 'Two', 'Three'], ordered=True)
+    df = pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], columns=pd.MultiIndex.from_arrays([["A", "B", "B"], columns]))
+    df.sum(axis=1, level=1)
```
Comment (dsaxton) [inline doc/source/user_guide/categorical.rst:627]: Can you format this to make it "black-like"?

Comment (phofl) [reply doc/source/user_guide/categorical.rst]: Yes, of course.

File: doc/source/user_guide/categorical.rst [hunk 2]
```diff
@@ -618,6 +618,19 @@ even if some categories are not present in the data:
     s = pd.Series(pd.Categorical(["a", "b", "c", "c"], categories=["c", "a", "b", "d"]))
     s.value_counts()
 
+``DataFrame`` methods like :meth:`DataFrame.sum` also show "unused" categories.
+
+.. ipython:: python
+
+    columns = pd.Categorical(
+            ["One", "One", "Two"], categories=["One", "Two", "Three"], ordered=True
+        )
```
Comment (dsaxton) [inline doc/source/user_guide/categorical.rst:627]:
```suggestion
    )
```

**Review by dsaxton — APPROVED** [review]

File: doc/source/user_guide/categorical.rst [hunk 3]
```diff
@@ -618,6 +618,19 @@ even if some categories are not present in the data:
     s = pd.Series(pd.Categorical(["a", "b", "c", "c"], categories=["c", "a", "b", "d"]))
     s.value_counts()
 
+``DataFrame`` methods like :meth:`DataFrame.sum` also show "unused" categories.
+
+.. ipython:: python
+
+    columns = pd.Categorical(
+            ["One", "One", "Two"], categories=["One", "Two", "Three"], ordered=True
+    )
```
Comment (dsaxton) [inline doc/source/user_guide/categorical.rst:627]:
```suggestion
    columns = pd.Categorical(
        ["One", "One", "Two"], categories=["One", "Two", "Three"], ordered=True
    )
```
Looks like one more RST lint error

Comment (phofl) [reply doc/source/user_guide/categorical.rst]: Yeah, I missed some indentations after running black... Stupid from me. Thx

**Comment by jreback** [comment]
sure, thanks

**Pull request merged**